autoload/neomake/makers/ft/sh.vim

```diff
@@ -40,6 +36,30 @@ function! neomake#makers#ft#sh#shellcheck() abort
     return maker
 endfunction
 
+function! neomake#makers#ft#sh#ShellcheckProcessJson(context) abort
+    let errors = []
+    for err in a:context.json
+        if err.level ==# 'info'
+            let type = 'I'
+        elseif err.level ==# 'warning'
+            let type = 'W'
+        else
+            let type = 'E'
+        endif
+
+        call add(errors, {
+            \ 'text': err.message,
+            \ 'lnum': err.line,
+            \ 'col': err.column,
+            \ 'length': err.endColumn - err.column,
+            \ 'filename': err.file,
```

Comment:
The `filename` part does not work with temporary files.
Likely setting/using `bufnr` instead would work / be better.

Reply:
This should be fixed via https://github.com/neomake/neomake/pull/2305.